Aromatics updates 1 (out of 4): Propargyl radical recombination to benzene

zjburas opened the pull request:
This is an updated version of #161 . All of the same changes apply. Briefly:

1. 8 new reaction families created to enable propargyl radical recombination to form benzene via elementary steps. Training reactions populated by 2003 Klippenstein calculations.
2. Created singlet and triplet carbene GAV's based on DFT_QCI_thermo library calculations, because singlet carbenes are critical intermediates in this pathway.
3. Created ring and polycyclic thermo groups for critical intermediates.

Also, there is a small, companion RMG-Py PR that goes along with this one in order to properly account for the new reaction families that are their own reverse: [#1142](https://github.com/ReactionMechanismGenerator/RMG-Py/pull/1142)

3 more pull requests are to come shortly that  build off of this one.

zjburas commented:
Only 4 of the database tests are being failed currently, but they are being propagated to the other 2 current Aromatics updates PR's, and I don't know how to pass them. @nyee or @mjohnson541 will probably have the best idea.

mliu49 commented:
In commit 0713dc0f9e0ca2b77bf14effb53d8c7c16c38f0f, you add one new thermo group, but there are multiple new tree entries. Travis is failing because it can't find the corresponding entries.

zjburas commented:
Thank you  @mliu49 . Yes, those orphaned ring() corrections in the tree were due to a shuffling of the commit history. Fixed now (and in subsequent PR's), but database tests will still fail due to 2 of the new reaction families not passing  unimolecular tests. I think I need to talk to either @nyee or @mjohnson541 to figure out the solution.

zjburas commented:
Thanks to @nyee and @mjohnson541, this PR, and PR's part 2 and 3, now passes all of the database tests. Whoever ends up reviewing these PR's please feel free to come and discuss with me.

zjburas commented on input/thermo/groups/polycyclic.py:
```diff
@@ -2517,9 +2517,9 @@
-        S298=(69.007, 'cal/(mol*K)'),
+        Cpdata=([-7.289, -6.524, -6.162, -6.421, -5.93, -4.094, -3.76], 'cal/(mol*K)'),
+        H298=(68.847, 'kcal/mol'),
+        S298=(65.182, 'cal/(mol*K)'),
```
The original s2_3_5_diene_0_2 ring() correction was sort of a placeholder, until I verified my thermo calculation. In this commit I update s2_3_5_diene_0_2 to the new calculated value that I was confident in. I have updated the commit message to clarify.

zjburas commented on input/kinetics/families/intra_H_migration/groups.py:
```diff
@@ -5238,3 +5238,23 @@
 
+##########
+forbidden(
+    label = "fulvene_H_mig_ring_edge_to_tail",
```
Yes, the other  forbidden group might have been accidentally deleted during conflict resolution while doing interactive rebase. I've added it back in.

zjburas commented on scripts/convertKineticsLibraryToTrainingReactions.ipynb:
```diff
@@ -104,21 +105,21 @@
+    "        rxts_cross = list(itertools.product(*rxt_resonances))\n",
+    "        rxts_cross = [list(cross) for cross in rxts_cross]\n",
+    "    else:\n",
+    "        rxts_cross = [[mol] for mol in lib_rxn.reactants[0].mol] #should have only one reactant\n",
```
Added this fix to [821771d](https://github.com/ReactionMechanismGenerator/RMG-database/pull/205/commits/82a771dc9d401a803068b63063692eb4fe81d32a)

zjburas commented on input/kinetics/libraries/2003_Miller_Propargyl_Recomb_High_P/reactions.py:
```diff
@@ -144,10 +144,28 @@
     label = "X <=> XI",
     degeneracy = 1,
-    kinetics = Arrhenius(A=(5.392e+10, 's^-1'), n=0.782, Ea=(34.256, 'kcal/mol'), T0=(1, 'K')),
+    kinetics = Arrhenius(A=(3.355e+12, 's^-1'), n=0.294, Ea=(35.954, 'kcal/mol'), T0=(1, 'K')),
```
I don't remember. All I can say is that the newer rate was used for the training reaction, which was in turned used in all subsequent validations tests (including comparison to Stein pyrolysis).

zjburas commented:
Thank you for the fast review, @mliu49 . In response to your comments I have done the following:

1. Deleted the redundant commit
2. Squashed the two pairs of commits you pointed out
3. Deleted unnecessary NIST folder
4. Addressed your other code-specific comments above

Regarding the drawbacks of removing singlet carbenes from the globally forbidden list: Previously there were only a couple of HBI corrections for singlet carbenes that  weren't even being accessed. We have verified that the new singlet carbene groups are being accessed, and although there is certainly room for improvement (for example, including a group for a secondary unsaturated carbene) the new groups will at least provide a reasonable estimate that can be refined further as needed. To the point, I've tested this branch against several systems (1,3-hexadiyne,  CPD and NG pyrolysis), and in none of the cases was the mechanism overrun with singlet carbenes. I did find a lot of species with either two carbene groups, or a carbene group on a radical, similar to how RMG will produce a lot of biradicals unless forbidden. I globally forbid such species in a later commit that is in PR #207 , perhaps this commit should be moved up?

zjburas commented:
As discussed, the commit forbidding species with two singlet carbene groups, or a singlet carbene and a radical, has been moved to this PR from PR #207 . 

The companion RMG-Py PR [#1142](https://github.com/ReactionMechanismGenerator/RMG-Py/pull/1142) needs unit tests, which will be done soon.
